rooms/views.py

```diff
@@ -1,3 +1,169 @@
+            search         = request.GET.get('search', '')           # 검색어
+            check_in       = request.GET.get('check_in', None)       # 체크인
+            check_out      = request.GET.get('check_out', None)      # 체크아웃
+
+            q = Q()
+
+            # 좌표의 범위값을 구분하여 좌표 사이에 있는 모든방 검색
+            if sw and ne:
+                latitude_top    = ne.split(',')[0]
+                latitude_bottom = sw.split(',')[0]
+                longitude_left  = sw.split(',')[1]
+                longitude_right = ne.split(',')[1]
+                
+                q &= Q(latitude__range = (latitude_top, latitude_bottom))
+                q &= Q(latitude__range = (longitude_left, longitude_right))
+
+            # 검색어가 주소일 경우
+            if search:
+                q &= Q(address__icontains = search)
+
+            # 성인, 어린이, 유아의 총합계로 max_people를 계산
+            if max_people:
+                q &= Q(max_people__gte = max_people)
+
+            # 예약가능일자 검색
+            if check_in and check_out:
+                q &= ~Q(Q(rooms_view__check_out_date__lt = check_out) & Q(rooms_view__check_in_date__gt = check_in))
+                q |= Q(rooms_view__check_in_date__isnull = True)
+
+            # 숙소유형
+            if room_type:
+                q &= Q(room_type_id__in = room_type)
+
+            # 침대 & 침대 개수
+            if bed and bed_count:
+                q &= Q(roomoptioninfo__room_option_id = bed)
+                q &= Q(roomoptioninfo__quantity = bed_count)
+
+            # 침실 & 침실 개수
+            if bedroom and bedroom_count:
+                q &= Q(roomoptioninfo__room_option_id = bedroom)
+                q &= Q(roomoptioninfo__quantity = bedroom_count)
+
+            # 욕실 & 욕실 개수
+            if bathroom and bathroom_count:
+                q &= Q(roomoptioninfo__room_option_id = bathroom)
+                q &= Q(roomoptioninfo__quantity = bathroom_count)
+
+            # 편의시설
+            if convenience:
+                q &= Q(roomconvenience__convenience_id__in = convenience)
+
+            # 요금     
+            q &= Q(price__range = (price_min, price_max))
+
+            room_options     = cache.get('room_options')
+            room_convenience = cache.get('room_convenience')
+
+            rooms = Room.objects.select_related('category', 'room_type') \
+                                .prefetch_related(
+                                    'rooms_view',
+                                    'roomoptioninfo_set', 
+                                    'image_set', 
+                                    'wish_set', 
+                                    'roomreview_set',
+                                    'roomconvenience_set') \
+                                .filter(q) \
+                                .order_by('-id')[((page - 1) * limit) : page * limit]
+                    
+            result = {
+                        'room_count' : rooms.aggregate(Count('id'))['id__count'],
```

Comment:
이미 가져온 값의 길이를 구하는 `len(rooms)`을 사용하지 않은 이유가 있으신가요?

```suggestion
                        'room_count' : rooms.aggregate(Count('id'))['id__count'],
```

Reply:
아 딱히 이유는 없고 먼가 orm으로 해결해 보고 싶은 소망이 있었습니다..ㅎ

Reply:
소망은 아무도 막을 수 없죠 ❤️❤️❤️